src/twisted/internet/tcp.py

```
@@ -337,7 +337,18 @@ def getTcpNoDelay(self):
         return bool(self.socket.getsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY))
 
     def setTcpNoDelay(self, enabled):
-        self.socket.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, enabled)
+        try:
+            # There are bug reports about failures when setting TCP_NODELAY under certain conditions
+            # on macOS: https://github.com/thespianpy/Thespian/issues/70,
+            # https://github.com/envoyproxy/envoy/issues/1446.
+            #
+            # It is reasonable to simply eat errors coming from setting TCP_NODELAY because
+            # TCP_NODELAY is relatively small performance optimization. In almost all cases the
+            # caller will not be able to do anything to remedy the situation and will simply
+            # continue.
+            self.socket.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, enabled)
+        except OSError:  # pragma: no cover
+            pass
```

Can we conditionally ignore the error only on macOS ?

I don't know under which conditions `TCP_NODELAY` might fail.

For a production code, I prefer to know that the failed.

Maybe emit a log message ?

---

> Can we conditionally ignore the error only on macOS ?

I think we should at least not fail. This would be endless source of frustration. Users will not be amused once they find that they spent hours tracking down rare failure (that may not have proper logs, as seen in the related Django bug), only to find that the cause is an optional optimization that twisted enabled.

Logging the failure is good idea though, thanks. I've implemented that solution.